### PR TITLE
feat: Added MultiSig DAO Dashboard and DAO Event Fetching

### PR DESCRIPTION
**Pull request by damiano-melcarne-lab49**
**Features**:
- Added Multis Sig DAO Dashboard
Screen Shot 2023-04-25 at 3 34 33 AM
Screen Shot 2023-04-25 at 3 34 47 AM
Screen Shot 2023-04-25 at 3 34 57 AM
Screen Shot 2023-04-25 at 3 35 08 AM
Screen Shot 2023-04-25 at 3 35 12 AM
Screen Shot 2023-04-25 at 3 35 27 AM
- Private DAOs are hidden in the UI
- Created HashscanLink component
Screen Shot 2023-04-25 at 3 52 25 AM
- Created Breadcrumb component
Screen Shot 2023-04-25 at 3 53 22 AM

**Tech Notes**
- Restructured the services to use DAO events + MirrorNode to fetch all DAO data.
- Created `useDAOTransactions` and `useAccountTokenBalances` hooks. Updated the `useDAOs` hook to filter by account Id.
- Applied `Page` layout to several other pages to keep spacing consistent in other sections of the app.
- Moved functions in `MirrorNodeService` to `DexService`. These are functions that do not directly map to the Mirror Node API endpoints.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
     "hashconnect": "^0.2.3",
     "immer": "^9.0.15",
     "numeral": "^2.0.6",
-    "ramda": "^0.28.0",
+    "ramda": "^0.29.0",
```
Comment (damiano-melcarne-lab49) [inline package.json:33]: Updated ramda to use the `swap` utility function.

File: src/dex-ui/pages/CreateProposal/CreateProposal.tsx [hunk 2]
```diff
@@ -48,37 +49,44 @@ export const CreateProposal = (props: CreateProposalProps) => {
...
-        icon={<WarningIcon color="#EF5C5C" h={10} w={10} />}
-        buttonConfig={{
-          text: "Dismiss",
-          onClick: () => {
-            governance.clearProposalTransactionState();
-          },
-        }}
-      />
-    </>
+    <Page
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/CreateProposal/CreateProposal.tsx:52]: Wrapped with `Page` layout, added `Breadcrumb` component.

File: src/dex-ui/pages/CreateProposal/SelectProposalType.tsx [hunk 3]
```diff
@@ -57,48 +55,48 @@ export const SelectProposalType = (props: any) => {
...
-        >
-          Continue
-        </Button>
-        <Spacer padding="0.25rem" />
-        <Button variant="cancel" width="437px" onClick={() => navigate(`/governance`)}>
-          Cancel
-        </Button>
-      </Center>
-    </VStack>
+    <Page
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/CreateProposal/SelectProposalType.tsx:58]: Wrapped with `Page` and `PageHeader` layout, added `Breadcrumb` component.

File: src/dex-ui/pages/Pools/Pools.tsx [hunk 4]
```diff
@@ -219,96 +219,98 @@ const Pools = (): JSX.Element => {
...
-          flexDirection={"column"}
-          index={poolsParamsState.selectedTab}
-          onChange={(index) => setPoolsParamsState({ ...poolsParamsState, selectedTab: index })}
-        >
-          {pools.withdrawTransactionState.status === "success" && poolsParamsState.showSuccessfulWithdrawalMessage ? (
-            <>
-              <Notification
-                type={NotficationTypes.SUCCESS}
-                message={`Withdrew ${pools.withdrawTransactionState.successPayload?.lpTokenAmount} LP Tokens from 
+        <Flex alignItems="center" padding="24px 80px 16px">
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/pages/Pools/Pools.tsx:222]: Temporary fix for Pool table padding.

File: src/dex-ui/services/DexService/DexService.ts [hunk 5]
```diff
@@ -28,93 +18,14 @@ const getTimeRemaining = (startBlock: string, endBlock: string): BigNumber => {
  * @returns
  */
 function createDexService() {
-  type ProposalData = MirrorNodeDecodedProposalEvent & ProposalDetailsResponse;
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/DexService/DexService.ts:31]: Moved to governance/GovernanceService

File: src/dex-ui/services/DexService/governance/GovernanceService.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+import { isNil } from "ramda";
+import { DexService } from "../..";
+import { ContractProposalState, Proposal, ProposalState, ProposalType } from "../../../store/governanceSlice";
+import { getStatus } from "../../../store/governanceSlice/utils";
+import { getTimeRemaining } from "../../../utils";
+import { MirrorNodeDecodedProposalEvent } from "../../MirrorNodeService";
+import { getFulfilledResultsData } from "../../MirrorNodeService/utils";
+import { Contracts, GovernanceTokenId } from "../../constants";
+import { ProposalData } from "./type";
+
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/DexService/governance/GovernanceService.ts:12]: Moved from MirrorNodeService

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 7]
```diff
@@ -162,28 +160,43 @@ function createMirrorNodeService() {
...
+  ): Promise<ethers.utils.LogDescription[]> {
+    const shouldParseAllEvents = isNil(events) || events.length === 0;
+    const logs: MirrorNodeEventLog[] = await fetchNextBatch(`/api/v1/contracts/${accountId}/results/logs`, "logs", {
+      params: {
+        order: "desc",
+      },
+    });
+    const parsedEventLogs: LogDescription[] = logs.reduce((logs: LogDescription[], log: MirrorNodeEventLog) => {
       try {
-        const data = log.data;
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:169]: Moved to utils

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 8]
```diff
@@ -224,68 +237,25 @@ function createMirrorNodeService() {
...
-    const allEvents = decodeLog(governorAbiSignatureMap, response.data.logs);
+    const allEvents = decodeLog(abiSignatures, response.data.logs, ["ProposalDetails"]);
     const proposalCreatedEvents = allEvents.get("ProposalDetails") ?? [];
     const proposals: MirrorNodeDecodedProposalEvent[] = proposalCreatedEvents.map((item: any) => {
       return { ...item, contractId, type: proposalType };
     });
     return proposals;
   };
 
-  /**
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:235]: Moved to DexService

File: src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts [hunk 9]
```diff
@@ -114,39 +112,39 @@ function createMirrorNodeService() {
...
-   * each token ID and formats the balances with the correct decimal positions.
-   * @param accountId - The ID of the account to return token balances for.
-   * @returns The list of balances (in decimal format) for the given account ID.
+   * Returns the results from a cost-free contract execution such as read-only smart contract queries,
+   * gas estimation, and transient simulation of read-write operations.
+   * @param payload -
+   * @returns Results from a cost-free EVM call to the contract.
    */
-  const fetchAccountTokenBalances = async (accountId: string): Promise<MirrorNodeAccountBalance> => {
-    const accountBalances = await fetchAccountBalances(accountId);
```
Comment (damiano-melcarne-lab49) [inline src/dex-ui/services/MirrorNodeService/MirrorNodeService.ts:130]: Moved to DexService

**Review by roshan-bisht-lab49 — APPROVED** [review]

**Pull request merged**